### PR TITLE
Fix/use default avatar lib

### PR DESCRIPTION
**Pull request by ulferts**
https://www.openproject.org/work_packages/13706
https://www.openproject.org/work_packages/14688
https://www.openproject.org/work_packages/14701
https://www.openproject.org/work_packages/14814
https://www.openproject.org/work_packages/14690

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://fix-use-default-avatar-lib.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

**Comment by myabc** [comment]
@ulferts did you want to rework this to use `gravatar_image_tag` everywhere instead?

**Comment by ulferts** [comment]
@myabc, I do and I already did so locally, so I will revert some of what I did.

File: app/assets/stylesheets/layout/_work_package.sass [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      margin-left: 5px
+
+  .issue
+    .attributes
+      td.work_package_attribute_header
+        font-weight: bold
+  .icon-button, .sort-header, .action-icon
+    cursor: pointer
+
+  select, input
```
Comment (ulferts) [inline app/assets/stylesheets/layout/_work_package.sass:42]: not sure if this is actually used any more

**Comment by ulferts** [comment]
@myabc, now it is the way we talked about.

File: spec/lib/api/v3/users/user_representer_spec.rb [hunk 2]
```diff
@@ -56,5 +59,34 @@
...
+      end
+
+      it 'should have an url to gravatar if settings permit and mail is set' do
+        expect(parse_json(subject, 'avatar')).to start_with('http://gravatar.com/avatar')
+      end
+
+      it 'should be nil if gravatar is disabled' do
+        Setting.stub(:gravatar_enabled?).and_return(false)
+
+        expect(parse_json(subject, 'avatar')).to be_nil
```
Comment (myabc) [inline spec/lib/api/v3/users/user_representer_spec.rb]: @ulferts `nil` or empty? – these examples are failing.

**Comment by ulferts** [comment]
@myabc, thanks for pointing out. Adapted the specs.

**Pull request merged**